commons/src/main/java/life/qbic/application/commons/Result.java

```diff
@@ -0,0 +1,242 @@
+package life.qbic.application.commons;
+
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * <b>Class Result</b>
+ *
+ * <p>This class introduces the Rust idiom to use Result objects to enforce the client code
+ * to apply some best practises to safe value access and proper exception handling.</p>
+ *
+ * <p>Results can be used to wrap an actual value of type <code>V</code>, which indicates a
+ * success, or an exception of type
+ * <code>E</code>. An object of type Result can only contain either a value or an exception, not
+ * both in the same instance.</p>
+ * <p>
+ * To properly deal with Result objects, a good idiom using Java's enhanced switch statements looks
+ * like this:
+ *
+ * <pre>
+ *
+ * final Result&lt;String, Exception&gt; result =
+ *          new Result("Contains actual information")
+ * // Using the  {@link Function} interface
+ * Function function = switch (result) {*    case result.isOk(): yield Function&lt;V, ?&gt;{...}*    case result.isError(): yield Function&lt;E, ?&gt;{...}*}* function.apply(result)
+ *
+ * // or using the {@link Consumer} interface
+ * Consumer consumer = switch (result) {*   case result.isOk() : yield Consumer&lt;V&gt;{...}*   case result.isError() : yield Consumer&lt;E&gt;{...}*}* consumer.accept(result)
+ *
+ * // or using lambda expressions
+ * switch (result) {*   case result.isOk() : () -> {}*   case result.isError() : () -> {}*}* </pre>
+ *
+ * @param <V> the value of this result in case it is OK
+ * @param <E> the type of error this result can hold
+ * @since 1.0.0
+ */
+class Result<V, E extends Exception> {
+
+  private final V value;
+  private final E exception;
+
+  private final Type type;
+
+  private enum Type {
+    FAILURE, SUCCESS
+  }
+
+  /**
+   * Static constructor method for creating a result object instance of type <code>V,E</code>
+   * wrapping an actual value <code>V</code>.
+   *
+   * @param value the notorious value to get wrapped in a result object
+   * @return a new result object instance
+   */
+  public static <V, E extends Exception> Result<V, E> success(V value) {
+    Objects.requireNonNull(value);
+    return new Result<>(value);
+  }
```

Comment:
```suggestion
  public static <V, E extends Exception> Result<V, E> value(V value) {
    Objects.requireNonNull(value);
    return new Result<>(value);
  }
```

Reply:
we create either a Result object of type success or failure